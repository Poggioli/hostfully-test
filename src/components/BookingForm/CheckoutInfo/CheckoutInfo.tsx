import { Separator } from "@/components/ui/separator";
import { FC } from "react";

type CheckoutInfoProps = {
    pricePerDay: number,
    bookingDays: number,
    totalPriceDay: number,
    pricePerGuest: number,
    quantityGuests: number,
    totalPriceGuest: number,
    totalPrice: number,
}

const CheckoutInfo: FC<CheckoutInfoProps> = ({
    pricePerDay,
    bookingDays,
    totalPriceDay,
    pricePerGuest,
    quantityGuests,
    totalPriceGuest,
    totalPrice
}) => {

    return (
        <div className="flex flex-col gap-4 mt-4">
            <h4 className="text-xl font-semibold tracking-tight">Booking informations</h4>
            <div className="flex gap-6 justify-between">
                <span className="underline">${pricePerDay} x {bookingDays} nights</span>
                <span>${totalPriceDay}</span>
            </div>
            <div className="flex gap-6 justify-between">
                <span className="underline">${pricePerGuest} x {quantityGuests} guests</span>
                <span>${totalPriceGuest}</span>
            </div>
            <Separator className="max-w-[200px] mx-auto" />
            <div className="flex gap-6 justify-between">
                <span className="text-lg font-semibold">Total</span>
                <span className="text-lg font-bold">${totalPrice}</span>
            </div>
        </div>
    );
}

export { CheckoutInfo };
