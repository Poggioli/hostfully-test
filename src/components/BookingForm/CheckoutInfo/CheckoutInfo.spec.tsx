import { render } from "@testing-library/react";
import { CheckoutInfo } from "./CheckoutInfo";

describe('CheckoutInfo', () => {
  it(`GIVEN a CheckoutInfo
      WHEN render
      THEN should render the props the right way`, () => {
    const { baseElement } = render(
      <CheckoutInfo
        bookingDays={5}
        pricePerDay={500}
        pricePerGuest={100}
        quantityGuests={3}
        totalPrice={2800}
        totalPriceDay={2500}
        totalPriceGuest={300}
      />
    )

    expect(baseElement).toMatchSnapshot();
  });
});