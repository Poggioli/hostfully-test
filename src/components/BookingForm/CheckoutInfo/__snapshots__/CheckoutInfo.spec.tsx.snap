// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutInfo GIVEN a CheckoutInfo
      WHEN render
      THEN should render the props the right way 1`] = `
<body>
  <div>
    <div
      class="flex flex-col gap-4 mt-4"
    >
      <h4
        class="text-xl font-semibold tracking-tight"
      >
        Booking informations
      </h4>
      <div
        class="flex gap-6 justify-between"
      >
        <span
          class="underline"
        >
          $
          500
           x 
          5
           nights
        </span>
        <span>
          $
          2500
        </span>
      </div>
      <div
        class="flex gap-6 justify-between"
      >
        <span
          class="underline"
        >
          $
          100
           x 
          3
           guests
        </span>
        <span>
          $
          300
        </span>
      </div>
      <div
        class="shrink-0 bg-border h-[1px] w-full max-w-[200px] mx-auto"
        data-orientation="horizontal"
        role="none"
      />
      <div
        class="flex gap-6 justify-between"
      >
        <span
          class="text-lg font-semibold"
        >
          Total
        </span>
        <span
          class="text-lg font-bold"
        >
          $
          2800
        </span>
      </div>
    </div>
  </div>
</body>
`;
